js/stores/windowStore.js

```diff
@@ -750,6 +752,22 @@ const doAction = (action) => {
         }
         break
       }
+    case windowConstants.WINDOW_ON_TOGGLE_BOOKMARK:
+      {
+        // show the AddEditBookmarkHanger control; saving/deleting takes place there
+        const activeFrame = frameStateUtil.getActiveFrame(windowState)
+        let siteDetail = siteUtil.getDetailFromFrame(activeFrame, siteTags.BOOKMARK)
+        const key = siteUtil.getSiteKey(siteDetail)
+
+        if (key !== null) {
```

Comment:
Here's a quick explanation of why this is a problem:
if you remove this line:
`siteDetail = siteDetail.set('parentFolderId', site.get('parentFolderId'))`

The code will create a NEW sites entry (which is wrong).  The reason why it creates a new one is because the key has the folderId in it.  The windowStore shouldn't care about this logic- it should just use the object it gets back.  Instead, we are having business logic in the windowStore about a siteDetails entry to avoid a bug (instead of returning an existing reference- the one which the key matches)